pkg/cdc/util.go

```diff
@@ -784,3 +784,78 @@ func addTailEndMetrics(bat *AtomicBatch) {
 	v2.CdcHoldChangesBytesGauge.Sub(allocated)
 	v2.CdcSinkRecordCounter.Add(count)
 }
+
+// uriHasPrefix
+func uriHasPrefix(uri string, prefix string) bool {
+	if len(uri) < len(prefix) || strings.ToLower(uri[:len(prefix)]) != prefix {
+		return false
+	}
+	return true
+}
+
+/*
+ExtractUriInfo extracts the uriInfo
+return serialized uriInfo
+*/
+func ExtractUriInfo(
+	ctx context.Context,
+	uri string,
+	uriPrefix string,
+) (string, UriInfo, error) {
+	ok, uriInfo := compositedUriInfo(uri, uriPrefix)
+	if !ok {
+		return "", UriInfo{}, moerr.NewInternalErrorf(ctx, "invalid uri format: %s", uri)
+	}
+
+	jsonUriInfo, err := JsonEncode(&uriInfo)
+	if err != nil {
+		return "", UriInfo{}, err
+	}
+	return jsonUriInfo, uriInfo, nil
+}
+
+// compositedUriInfo uri according to the format: mysql://root:111@127.0.0.1:6001
+// if valid, return true and extracted info
+// !!!NOTE!!!
+// user and password does not have the special character ( ':' '@' )
```

Comment:
Is this enforced somehow/somewhere?    Otherwise seems to me, the only way is to encode/decode this with &XXX; 
Or some better/smarter encode/decode scheme.